garrysmod/lua/includes/extensions/net.lua

```diff
@@ -106,32 +106,61 @@ end
 -- item indivdually and in a specific order
 -- because it adds type information before each var
 --
-function net.WriteTable( tab )
+function net.WriteTable( tab, seq )
 
-	for k, v in pairs( tab ) do
+	if ( seq ) then
 
-		net.WriteType( k )
-		net.WriteType( v )
+		local len = #tab
+		net.WriteUInt( len, 32 )
 
-	end
+		for i = 1, len do
+
+			net.WriteType( tab[i] )
+
+		end
+		
+	else
 
-	-- End of table
-	net.WriteType( nil )
+		for k, v in pairs( tab ) do
+	
+			net.WriteType( k )
+			net.WriteType( v )
+	
+		end
+	
+		-- End of table
+		net.WriteType( nil )
+
+	end
 
 end
 
-function net.ReadTable()
+function net.ReadTable( seq )
 
 	local tab = {}
 
-	while true do
+	if ( seq ) then
+
+		for i = 1, net.ReadUInt(32) do
+
+			tab[i] = net.ReadType()
+
+		end
+
+	else
+
+		while true do
+
+			local k = net.ReadType()
+			if ( k == nil ) then return tab end
```

Comment:
Change this to break. Makes the return statement unified between the two conditional branches which eases reading.

Reply:
Had the same feeling